### PR TITLE
task1 fixes (async, try/catch)

### PR DESCRIPTION
**Pull request by StellarCrow**

### CODE REVIEW

File: task1/server.js [hunk 1]
```diff
@@ -13,61 +17,57 @@ http
...
+    catch(err) {
+      console.error(err);
+      return;
     }
 
     res.writeHead(200, { "Content-Type": "application/json" });
 
     if (req.url === "/get-all-logs" && req.method === "GET") {
-      return res.end(JSON.stringify(readFromFile(path)));
+      return res.end(JSON.stringify(await readFromFile(path)));
```
Comment (staff2708) [inline task1/server.js]: use try catch for json stringify

File: task1/server.js [hunk 2]
```diff
@@ -13,61 +17,57 @@ http
...
     }
 
-    let urlHasQuery = (Object.keys(queryObject).length === 0) ? false : true;
+    let urlHasQuery = !(Object.keys(queryObject).length === 0);
```
Comment (staff2708) [inline task1/server.js]: maybe its better to do smth like this Object.keys(queryObject).length !== 0

File: task1/server.js [hunk 3]
```diff
@@ -13,61 +17,57 @@ http
...
     if (urlHasQuery) {
       const from = queryObject.from;
       const to = queryObject.to;
-      let array = getDataRanged(path, from, to);
+      let array = await getDataRanged(path, from, to);
```
Comment (staff2708) [inline task1/server.js]: use const instead of let

Comment (staff2708) [reply task1/server.js]: also give this variable some more meaningful name

File: task1/server.js [hunk 4]
```diff
@@ -13,61 +17,57 @@ http
...
     }
 
     res.end(JSON.stringify({ status: "ok" }));
   })
   .listen(PORT);
 
-function writeToJson(path, requestInfo) {
-  let data = readFromFile(path);
+async function writeToJson(path, requestInfo) {
+  let data = await readFromFile(path);
```
Comment (staff2708) [inline task1/server.js]: const

File: task1/server.js [hunk 5]
```diff
@@ -13,61 +17,57 @@ http
...
+    return JSON.parse(data);
+  } catch (err) {
+    throw err;
+  }
 }
 
-function getDataRanged(path, from, to) {
-  let data = readFromFile(path);
+async function getDataRanged(path, from, to) {
+  let data = await readFromFile(path);
```
Comment (staff2708) [inline task1/server.js]: const

File: task1/server.js [hunk 6]
```diff
@@ -13,61 +17,57 @@ http
...
   return data.logs.filter(elem => {
     let date = elem.time.toString().split("T")[0];
     if (date >= from && date <= to) return elem;
```
Comment (staff2708) [inline task1/server.js]: maybe its better to do smth like this 
`data.logs.filter(log => {
  const logDate = log.time.toString().split("T")[0];
  return logDate >= from && logDate <= to
})`

File: task1/server.js [hunk 7]
```diff
@@ -13,61 +17,57 @@ http
...
-    if (fs.existsSync(path)) {
-      writeToJson(path, requestInfo);
-    } else {
-      createNewFile(path, requestInfo);
+    try {
+      await writeToJson(path, requestInfo);
+    }
+    catch(err) {
+      console.error(err);
+      return;
```
Comment (staff2708) [inline task1/server.js]: its not good, because, if we do this, user will wait until response,  till request timeout

File: task1/server.js [hunk 8]
```diff
@@ -13,61 +17,57 @@ http
...
     }
 
-    let urlHasQuery = (Object.keys(queryObject).length === 0) ? false : true;
+    let urlHasQuery = !(Object.keys(queryObject).length === 0);
     if (urlHasQuery) {
       const from = queryObject.from;
       const to = queryObject.to;
-      let array = getDataRanged(path, from, to);
+      let array = await getDataRanged(path, from, to);
       return res.end(JSON.stringify(array));
```
Comment (staff2708) [inline task1/server.js]: im not sure, but shouldnt  we send logs only if its /get-all-logs route?

**Review by staff2708 — COMMENTED** [review]
Few comments

File: task1/server.js (hunk 8 above)
Comment (StellarCrow) [reply task1/server.js]: No, we should have a route for getting logs for specific date range.

File: task1/server.js [hunk 9]
```diff
@@ -13,61 +17,62 @@ http
...
+  }
 }
 
-function readFromFile(path) {
-  let data = fs.readFileSync(path, err => {
-    if (err) throw err;
+async function getDataRanged(path, from, to) {
+  const data = await readFromFile(path);
+  return data.logs.filter(log => {
+    let logDate = log.time.toString().split("T")[0];
```
Comment (staff2708) [inline task1/server.js]: there still should be const :)

File: task1/server.js [hunk 10]
```diff
@@ -13,61 +17,62 @@ http
...
 
     res.writeHead(200, { "Content-Type": "application/json" });
 
     if (req.url === "/get-all-logs" && req.method === "GET") {
-      return res.end(JSON.stringify(readFromFile(path)));
+      try {
+        const allLogs = JSON.stringify(await readFromFile(path));
+        return res.end(allLogs);
+      } catch(err) {
+        throw err;
```
Comment (staff2708) [inline task1/server.js:37]: if we just throw error, but dont send response to user, user will be upset

**Review by staff2708 — APPROVED** [review]

**Pull request merged**